pkg/serverless/invocationlifecycle/lifecycle.go

```diff
@@ -230,12 +231,21 @@ func (lp *LifecycleProcessor) OnInvokeStart(startDetails *InvocationStartDetails
 		}
 		ev = event
 		lp.initFromLambdaFunctionURLEvent(event, region, account, resource)
+	case trigger.StepFunctionEvent:
+		var event events.StepFunctionEvent
+		if err := json.Unmarshal(payloadBytes, &event); err != nil && arnParseErr != nil {
```

Comment:
You won't need to check `arnParseErr` here since you're not using anything of the variables (`region`, `account`, `resource`) parsed out from the arn

Reply:
Maximum clever